syntax = "proto3";

package lars;

/* Lars系统的消息ID */
enum MessageId {
  ID_UNKNOW = 0;            // 占位
  ID_GetRouterRequest = 1;  // 向DNS请求Router对应的关系的消息ID
  ID_GetRouterResponse = 2; // DNS回复给Agent结果的消息
}

/* 主机的信息 */
message HostInfo {
  int32 ip = 1;
  int32 port = 2;
}

/* 请求dns路由信息的消息 */
message GetRouterRequest {
  int32 modid = 1;
  int32 cmdid = 2;
}

/* DNS回复的消息 */
message GetRouterResponse {
  int32 modid = 1;
  int32 cmdid = 2;
  repeated HostInfo hosts = 3;
}